src/plugins/orangehrmCoreOAuthPlugin/Api/OAuthClientAPI.php

```diff
@@ -201,42 +304,41 @@ public function update(): EndpointResult
     public function getValidationRuleForUpdate(): ParamRuleCollection
     {
         return new ParamRuleCollection(
+            new ParamRule(CommonParams::PARAMETER_ID, new Rule(Rules::POSITIVE)),
             ...$this->getCommonBodyValidationRules(),
         );
     }
 
     /**
-     * @return OAuthClient
-     * @throws RecordNotFoundException|DaoException
+     * @param OAuthClient $oAuthClient
      */
-    public function saveOAuthClient(): OAuthClient
+    public function setOAuthClient(OAuthClient $oAuthClient): void
     {
-        $clientId = $this->getRequestParams()->getString(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_CLIENT_ID);
-        $existingClientId = $this->getRequestParams()->getString(
-            RequestParams::PARAM_TYPE_QUERY,
-            self::PARAMETER_CLIENT_ID
+        $oAuthClient->setName(
+            $this->getRequestParams()->getString(
+                RequestParams::PARAM_TYPE_BODY,
+                self::PARAMETER_NAME
+            )
         );
-        $clientSecret = $this->getRequestParams()->getString(
-            RequestParams::PARAM_TYPE_BODY,
-            self::PARAMETER_CLIENT_SECRET
+
+        $oAuthClient->setRedirectUri(
+            $this->getRequestParams()->getString(
+                RequestParams::PARAM_TYPE_BODY,
+                self::PARAMETER_REDIRECT_URI
+            )
         );
-        $redirectUri = $this->getRequestParams()->getString(
-            RequestParams::PARAM_TYPE_BODY,
-            self::PARAMETER_REDIRECT_URI
+
+        $oAuthClient->setEnabled(
+            $this->getRequestParams()->getBooleanOrNull(
+                RequestParams::PARAM_TYPE_BODY,
+                self::PARAMETER_ENABLED
+            )
         );
-        if (!empty($existingClientId)) {
-            $oAuthClient = $this->getOAuthService()->getOAuthClientByClientId($existingClientId);
-            $this->throwRecordNotFoundExceptionIfNotExist($oAuthClient, OAuthClient::class);
-        } else {
-            $oAuthClient = new OAuthClient();
-            // TODO
-            $oAuthClient->setConfidential(GrantType::CLIENT_CREDENTIALS);
-            $oAuthClient->setScope(Scope::SCOPE_ADMIN);
-        }
-
-        $oAuthClient->setName($clientId);
-        $oAuthClient->setClientSecret($clientSecret);
-        $oAuthClient->setRedirectUri($redirectUri);
-        return $this->getOAuthService()->saveOAuthClient($oAuthClient);
+
+        // TODO client secret, client id generation
+        $oAuthClient->setClientId(bin2hex(random_bytes(32)));
+        $oAuthClient->setClientSecret(bin2hex(random_bytes(32)));
```

Comment:
TODO:: increase entropy/length for the secret generation

Reply:
moved this generation into create function. otherwise id and secret generating again for update also